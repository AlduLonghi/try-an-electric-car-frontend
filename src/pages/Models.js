import { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Carousel from 'react-multi-carousel';
import fetchCars from '../redux/actions/cars';
import Navbar from '../components/Navbar';
import CarLink from '../components/CarLink';
import 'react-multi-carousel/lib/styles.css';
import '../styles/car-link.scss';

const Models = ({ fetchCars, cars }) => {
  useEffect(() => {
    fetchCars();
  }, []);

  const responsive = {
    superLargeDesktop: {
      breakpoint: { max: 3000, min: 1024 },
      items: 3,
    },
    desktop: {
      breakpoint: { max: 1024, min: 764 },
      items: 2,
    },
    tablet: {
      breakpoint: { max: 764, min: 464 },
      items: 1,
    },
    mobile: {
      breakpoint: { max: 464, min: 0 },
      items: 1,
    },
  };

  let toRenderComponent;
  if (cars.length !== 0) {
    toRenderComponent = (
      <Carousel className="mx-auto carousel-cont" responsive={responsive} itemClass="carousel-item-padding-40-px">
        <CarLink model={cars[0].model} pic={cars[0].profPic[0].url} />
        <CarLink model={cars[1].model} pic={cars[1].profPic[0].url} />
        <CarLink model={cars[2].model} pic={cars[2].profPic[0].url} />
        <CarLink model={cars[3].model} pic={cars[3].profPic[0].url} />
        <CarLink model={cars[4].model} pic={cars[4].profPic[0].url} />
      </Carousel>
    );
  } else {
    toRenderComponent = 'hello';
  }

  return (
    <div className="h-100">
      <Navbar />
      <main className="main-models">
        {toRenderComponent}
      </main>
    </div>
  );
};

const mapStateToProps = state => ({
  cars: state.cars.cars,
});

const mapDispatchToProps = {
  fetchCars,
};

Models.propTypes = {
  cars: PropTypes.instanceOf(Array).isRequired,
  fetchCars: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, mapDispatchToProps)(Models);
