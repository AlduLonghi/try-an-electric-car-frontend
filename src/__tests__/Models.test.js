import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import Models from '../pages/Models';
import store from '../redux/store';

it('renders Models page correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={['/models']}>
        <Models />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
