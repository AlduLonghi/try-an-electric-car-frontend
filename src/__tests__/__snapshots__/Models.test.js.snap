// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Models page correctly 1`] = `
<div
  className="models-cont"
>
  <div>
    <div>
      <div
        className="collapse-btn"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          <span
            className="material-icons"
          >
            drag_handle
          </span>
        </button>
        <button
          className="logout-btn text-center display-block"
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </div>
    </div>
    <nav
      className="display-none"
    >
      <h1
        className="text-center"
      >
        Electricar
      </h1>
      <ul
        className="text-uppercase font-weight-bold py-1 text-center d-flex flex-column justify-content-center"
      >
        <li>
          <a
            aria-current={null}
            className="a-navbar"
            href="/lifestyle"
            onClick={[Function]}
          >
            lifestyle
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            className="a-navbar active-navlink"
            href="/models"
            onClick={[Function]}
            style={Object {}}
          >
            models
          </a>
        </li>
        <li>
          <a
            aria-current={null}
            className="a-navbar"
            href="/profile"
            onClick={[Function]}
          >
            profile
          </a>
        </li>
      </ul>
      <button
        className="logout-btn text-center display-block"
        onClick={[Function]}
        type="button"
      >
        Logout
      </button>
      <footer
        className="as-navbar-component"
      >
        <div
          className="d-flex justify-content-around w-50 mx-auto mb-2"
        >
          <a
            href="https://github.com/AlduLonghi"
          >
            <img
              alt="github"
              src="github.svg"
            />
          </a>
          <a
            href="https://www.linkedin.com/in/aldanalonghi/"
          >
            <img
              alt="linkedin"
              src="linkedin.svg"
            />
          </a>
          <a
            href="https://www.behance.net/gallery/26425031/Vespa-Responsive-Redesign"
          >
            <img
              alt="behance"
              src="behance.svg"
            />
          </a>
        </div>
        <p
          className="text-center p-footer"
        >
          Â©2021 Aldana Longhi
        </p>
      </footer>
    </nav>
  </div>
  <main
    className="main-models"
  >
    <div
      className="w-100 h-100 d-flex align-content-center"
    >
      <div
        className="loading loading4"
      />
    </div>
  </main>
</div>
`;
